crates/nu-protocol/src/signature.rs

```diff
@@ -610,6 +610,19 @@ impl Signature {
             block_id,
         })
     }
+
+    pub fn formatted_flags(self) -> String {
+        let mut s = "Available flags:".to_string();
+        for flag in self.named {
+            if let Some(short) = flag.short {
+                s = format!("{s} --{}(-{}),", flag.long, short);
```

Comment:
Please use [`write!()`](https://doc.rust-lang.org/std/macro.write.html) instead as we can explicitly reuse the buffer. There is no need to include the old buffer in the format string.

Alternatively you can use the `FromIterator<String>`/`FromIterator<&`a str>` implementation and use `.collect()`